closed/adds/jdk/src/share/native/jdk/crypto/jniprovider/NativeCrypto.c

```diff
@@ -96,22 +85,41 @@ JNIEXPORT jlong JNICALL Java_jdk_crypto_jniprovider_NativeCrypto_DigestCreateCon
             digestAlg = EVP_sha512();
             break;
         default:
-            assert(0);
+            return -1;
     }
 
-    if((ctx = EVP_MD_CTX_new()) == NULL)
-        handleErrors();
+    if ((ctx = EVP_MD_CTX_new()) == NULL) {
+        printErrors();
+        return -1;
+    }
 
-    if(1 != EVP_DigestInit_ex(ctx, digestAlg, NULL))
-        handleErrors();
+    if (1 != EVP_DigestInit_ex(ctx, digestAlg, NULL)) {
+        printErrors();
+        EVP_MD_CTX_free(ctx);
+        return -1;
+    }
 
     context = malloc(sizeof(OpenSSLMDContext));
+    if (context == NULL) {
+        EVP_MD_CTX_free(ctx);
+        return -1;
+    }
     context->ctx = ctx;
     context->digestAlg = digestAlg;
 
     if (copyContext != 0) {
         EVP_MD_CTX *contextToCopy = ((OpenSSLMDContext*) copyContext)->ctx;
```

Comment:
The cast here is likely to elicit a warning from compilers because `jlong` may be a different size than a pointer; it should be cast to `intptr_t` first, like this:
```
EVP_MD_CTX *contextToCopy = ((OpenSSLMDContext *)(intptr_t)copyContext)->ctx;
```